experimental/ast/expr_dict.go

```diff
@@ -23,6 +23,16 @@ import (
 )
 
 // ExprDict represents a an array of message fields between curly braces.
+//
+// ExprArray implements [Commas].
+//
+// # Grammar
+//
+//	ExprDict := `{` fields `}` | `<` fields `>`
+//	fields := (Expr [,;]?)* Expr
```

Comment:
We _have_ to support trailing separator here since the protobuf language does. Also, empty braces are fine and mean an empty message, whereas these productions imply at least one element is required.

Also, I think it's more clear to stick with backticks to represent literal text sequences and then `|` for alternatives, instead of the more regex-y `[...]`.

Also, I suspect the inconsistent case is likely an accident. Let's stick with PascalCase. Though it would be nice to use snake_case for lexical elements (productions not defined here... which we might want to define in the token package?)

```suggestion
//	ExprDict := `{` Fields `}` | `<` Fields `>`
//	Fields := ( Expr ( `,` | `;` )? )?
```

Reply:
Agreed re: character clases.

The `camelCase` is intentional, the `doc.go` update explains it.

Let me update all the lists to show they do parse a trailing delimiter (even if it's disallowed).